subprojects/core/src/main/java/org/gradle/api/internal/plugins/PluginTargetType.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.plugins;
+
+import org.gradle.api.NonNullApi;
+import org.gradle.api.Project;
+import org.gradle.api.initialization.Settings;
+import org.gradle.api.invocation.Gradle;
+
+import javax.annotation.Nullable;
+
+@NonNullApi
+public enum PluginTargetType {
+
+    PROJECT("a Project (or in a build script)"),
+    SETTINGS("the Settings (or in a settings script)"),
```

Comment:
💭 `the` reads oddly here and in context but I'm not a native speaker. I'd remove it and simply have `Settings ...` and `Gradle ...`. Alternatively, `the Settings *object*` and `the Gradle *object*` like it's done elsewhere also look good.